src/Illuminate/Database/Eloquent/Factories/Factory.php

```diff
@@ -591,11 +591,13 @@ public function hasAttached($factory, $pivot = [], $relationship = null)
             'has' => $this->has->concat([new BelongsToManyRelationship(
                 $factory,
                 $pivot,
-                $relationship ?? Str::camel(Str::plural(class_basename(
+                $relationship ?? Str::of(class_basename(
```

Comment:
You could use `Str::of(...)->classBasename()`

https://laravel.com/docs/9.x/helpers#method-fluent-str-class-basename
(There are also some other places)